src/models/user.js

```diff
@@ -23,7 +23,7 @@ export default (sequelize, DataTypes) => {
     },
     salt: {
       type: DataTypes.STRING,
-      allowNull: false
+      allowNull: true
```

Comment:
Can the `salt` be `null`? How do you encrypt the password if it's `null`?

Reply:
Since the salt is generated randomly, I thought that the user didn't have to pass one in the parameters for creating a User. That's why even if it's null when a user is passing the arguments, it will create a salt automatically because the password must not be null and so the salt will never be null after the user is created.

Reply:
Yes, you're right. But you're confusing two layers here: GraphQL vs Sequelize.
- At the database layer (Sequelize), you have to think how you want your data to be stored, independently of how that data gets inserted in the database. If all users have to have a password **and** a salt, then both fields have to be non-null.
- At the input layer (GraphQL), you validate the data that you're receiving from an external source. Here, you don't need the salt. As you said, the salt is generated each time a password is changed.

It looks like this:
```
     GraphQL     |       Sequelize model     |    storage of the data
-----------------------------------------------------------------------------------
new password --> new password + salt --> database
```

In the Sequelize model layer you add the salt, because every user **has** to have one associated. **The database layer enforces data integrity**, so you _have_ to tell the database that you want every user to have a salt. That way, if you forget to add the code that adds the salt, the database will not allow you to insert the row, because if the data integrity rules.

Reply:
Okay, I understand. Thanks for the explanation!